presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -168,6 +172,7 @@ private SqlQueryExecution(
             this.schedulerStats = requireNonNull(schedulerStats, "schedulerStats is null");
             this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
             this.costCalculator = requireNonNull(costCalculator, "costCalculator is null");
+            this.dynamicFilterService = requireNonNull(dynamicFilterService, "dynamicFilterService is null");
```

Comment:
add a dedicate state change listener here for dynamic filtering that would:
* register
* deregister query when it's done

Reply:
done